linode/manager
# feat: [M3-7086] - Add pricing Docs Link to create/clone flows and remove DC-specific pricing warning notice

## Description 📝
Post-launch and after two full billing cycles, we are going to remove the info notice below the region selector that appears when a region with DC-specific pricing (currently: Jakarta; Sao Paulo) is selected and add a docs link "How Datacenter Pricing Works". The docs link was present in initial mocks, but determined to be redundant while the notice with "Learn more" existed. 

## Changes  🔄
- Removed the warning notice when a DC-specific pricing region is selected.
- Added the permanent docs link to the pricing page in the region select section of Linode/Kube/NodeBalancer Create (and Linode Clone) flows.
   -  In Kube Create, UX has signed off on placement of the docs link and moving the speedtext helper text below the region select. This looked like the cleanest placement for a form that doesn't quite match the others, since it is all in one paper and lacks a panel header.
- Cleaned up test integration and unit test coverage to test for the correct docs link, not the notice.

## Preview 📷
| Location | Before  | After   |
| ------- |------- | ------- |
| Linode Create / Distros | ![Screenshot 2023-12-01 at 1 08 35 PM](https://github.com/linode/manager/assets/114685994/0aa99d81-6aec-4781-a41c-8ffa6fdf75a8) | ![Screenshot 2023-12-01 at 1 09 11 PM](https://github.com/linode/manager/assets/114685994/958b515d-2a43-4671-b878-caa9e66df656) |
| Linode Create / Clone from DC-specifc-pricing linode | ![Screenshot 2023-12-01 at 1 15 38 PM](https://github.com/linode/manager/assets/114685994/1f68fa84-a1d0-4ca3-b66f-e24276e67380) | ![Screenshot 2023-12-01 at 1 15 47 PM](https://github.com/linode/manager/assets/114685994/eee3340c-4109-4051-b970-f68d502ec67e) |
| Linode Create / Clone from non-DC-specific-pricing linode but change the region | ![Screenshot 2023-12-01 at 1 19 12 PM](https://github.com/linode/manager/assets/114685994/724ee74d-da95-45a4-9d7f-fb71fd5dfaa8) | ![Screenshot 2023-12-01 at 1 20 28 PM](https://github.com/linode/manager/assets/114685994/8f2934c1-2980-4a9d-9f8a-f35a60949cd7) |
| NodeBalancer Create  | ![Screenshot 2023-12-01 at 1 01 09 PM](https://github.com/linode/manager/assets/114685994/c4caea45-ec65-489e-81bb-67871c0681f6) | ![Screenshot 2023-12-01 at 1 00 46 PM](https://github.com/linode/manager/assets/114685994/c986f68e-8a7e-44f9-86a7-80d5da14051b) |
| Kube Cluster Create  | ![Screenshot 2023-12-01 at 1 10 04 PM](https://github.com/linode/manager/assets/114685994/f86b9c2f-73b8-4343-9f0a-0407717b862c) | ![Screenshot 2023-12-01 at 1 09 42 PM](https://github.com/linode/manager/assets/114685994/1ccc1a43-7f93-4905-b7bd-4a936e228cb6) |

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Create a linode in Jakarta or Sao Paulo, if you do not have one.
- Create a linode in anywhere other than Jakarta or Sao Paolo, if you do not have one. 
- Check out this PR and `yarn up`.

### Verification steps 
(How to verify changes)
- Go to each of the create pages shown in the Preview above.
   - Confirm that you *do* see a Docs Link to the right of the region select that says `How Data Center Pricing Works` and links to https://www.linode.com/pricing.
   - Select Jakarta or Sao Paulo from the dropdown and confirm you do not see a warning notice that says `Prices for plans, products, and services in ... may vary from other regions. Learn more.`
- For the Clone flow, confirm:
   - If you are cloning from a Jakarta or Sao Paulo linode (the one you made in the Prerequisite step), you should not see the `Prices for...` notice.
   - If you are cloning from a non-DC-specific-pricing linode (e.g. located in Seattle) and you change the region to Jakarta or Sao Paulo, you *should* still see the `The selected region has a different price structure. Learn more.` notice. This is a situation where users will incur different charges for that linode if they proceed, so we want to keep this notice.
- Ensure the following tests pass in CI (and test them locally, if you'd like):

Cypress:
```
yarn cy:run -s "cypress/e2e/core/linodes/clone-linode.spec.ts,cypress/e2e/core/linodes/create-linode.spec.ts,cypress/e2e/core/nodebalancers/smoke-create-nodebal.spec.ts,cypress/e2e/core/kubernetes/lke-create.spec.ts"
```

Unit:
```
yarn test SelectRegionPanel
```

## As an Author I have considered 🤔

*Check all that apply*
